src/components/Table/EditableDraggableTable.js

```diff
@@ -96,17 +97,24 @@ class EditableDraggableTable extends React.Component {
     onCancel: ({ rowData }) => {
       const editing = false;
       const { id } = rowData;
-      const rows = cloneDeep(this.props.rows);
-      const index = findIndex(rows, { id });
-
-      rows[index] = cloneDeep(rows[index].backup);
-      delete rows[index].backup;
+      const index = findIndex(this.props.rows, { id });
+      let rows;
+
+      let type = ON_CANCEL;
+      if (this.props.rows[index].newRow) {
+        rows = this.props.rows.filter(row => row.id !== id);
```

Comment:
can't we rather call onDelete method from here?

Reply:
extracted onDelete to separate function which is now reused where needed